monkey/common/types/concurrency.py

```diff
@@ -0,0 +1,40 @@
+from types import TracebackType
+from typing import Optional, Type
+
+from typing_extensions import Protocol
+
+
+class Lock(Protocol):
+    def __enter__(self) -> bool:
+        ...
+
+    def __exit__(
+        self,
+        exc_type: Optional[Type[BaseException]],
+        exc_val: Optional[BaseException],
+        exc_tb: Optional[TracebackType],
```

Comment:
```suggestion
        exception_type: Optional[Type[BaseException]],
        exception_val: Optional[BaseException],
        exception_tb: Optional[TracebackType],
```

Reply:
The names here are consistent with what's in the python documentation. For that reason, I wouldn't change them